### PR TITLE
O3-2896: Appointment - Allow scheduling / editing Provider from Appoi…

### PR DESCRIPTION
**Pull request by mogoodrich**
…ntment Form

## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

### CODE REVIEW

File: packages/esm-appointments-app/src/form/appointments-form.component.tsx [hunk 1]
```diff
@@ -128,6 +131,10 @@ const AppointmentsForm: React.FC<AppointmentsFormProps> = ({
     resolver: zodResolver(appointmentsFormSchema),
     defaultValues: {
       location: appointment?.location?.uuid ?? session?.sessionLocation?.uuid ?? '',
+      provider:
+        appointment?.providers?.find((provider) => provider.response === 'ACCEPTED')?.uuid ??
+        session?.currentProvider.uuid ??
+        '', // assumes only a single provider, if multiple, just takes the first
```
Comment (mogoodrich) [inline packages/esm-appointments-app/src/form/appointments-form.component.tsx]: So it looks like the  appointments API the "provider" reference on appointment has been replaced by a patient_appointments_provider table to allow multiple providers (though, tangent, the provider column has not been removed though it doesn't seem to be used anymore). When the provider is changed, the old provider is marked as "CANCELLED".  The current provider is the one marked as "ACCEPTED".  I believe that this (repopulating the provider on edit) didn't work with the old code.

File: packages/esm-appointments-app/src/form/appointments-form.component.tsx [hunk 2]
```diff
@@ -128,6 +131,10 @@ const AppointmentsForm: React.FC<AppointmentsFormProps> = ({
     resolver: zodResolver(appointmentsFormSchema),
     defaultValues: {
       location: appointment?.location?.uuid ?? session?.sessionLocation?.uuid ?? '',
+      provider:
+        appointment?.providers?.find((provider) => provider.response === 'ACCEPTED')?.uuid ??
+        session?.currentProvider.uuid ??
+        '', // assumes only a single previously-scheduled provider with state "ACCEPTED", if multiple, just takes the first
```
Comment (mogoodrich) [inline packages/esm-appointments-app/src/form/appointments-form.component.tsx:137]: So it looks like the appointments API the "provider" reference on appointment has been replaced by a patient_appointments_provider table to allow multiple providers (though, tangent, the provider column has not been removed though it doesn't seem to be used anymore). When the provider is changed, the old provider is marked as "CANCELLED". The current provider is the one marked as "ACCEPTED". I believe that this (repopulating the provider on edit) didn't work with the old code.

File: packages/esm-appointments-app/src/form/appointments-form.component.tsx [hunk 3]
```diff
@@ -128,6 +131,10 @@ const AppointmentsForm: React.FC<AppointmentsFormProps> = ({
     resolver: zodResolver(appointmentsFormSchema),
     defaultValues: {
       location: appointment?.location?.uuid ?? session?.sessionLocation?.uuid ?? '',
+      provider:
+        appointment?.providers?.find((provider) => provider.response === 'ACCEPTED')?.uuid ??
+        session?.currentProvider.uuid ??
```
Comment (mseaton) [inline packages/esm-appointments-app/src/form/appointments-form.component.tsx:136]: What if there is no currentProvider?

**Review by mseaton — APPROVED** [review]
Generally looks ok I think.  Basically, if it works, I'm good with it.  I did add a few comments to check on.

File: packages/esm-appointments-app/src/form/appointments-form.component.tsx [hunk 4]
```diff
@@ -620,6 +627,36 @@ const AppointmentsForm: React.FC<AppointmentsFormProps> = ({
...
+          <span className={styles.heading}>{t('provider', 'Provider')}</span>
+          <div>
+            <ResponsiveWrapper>
+              <Controller
+                name="provider"
+                control={control}
+                render={({ field: { onChange, value, onBlur, ref } }) => (
+                  <Select
+                    id="provider"
+                    invalidText="Required"
```
[the remark this answers is not in the document]
Comment (mogoodrich) [reply packages/esm-appointments-app/src/form/appointments-form.component.tsx:640]: I just stole this from other selects in this form, but, yeah, it clearly looks like text should be localized.  That being said, I couldn't find a way to actually display this text (if you skip a required field and hit save it just highlights the required field in red).

So going to merge this in as-is, but happy to either 1) convert the Required to a t function or 2) remove the attribute entirely... @ibacher  @denniskigen  do you know if/how this text ends up rendering?

**Pull request merged**